src/network/core/packet.cpp

```diff
@@ -397,6 +396,34 @@ void Packet::Recv_string(char *buffer, size_t size, StringValidationSettings set
 	str_validate(bufp, last, settings);
 }
 
+/**
+ * Reads characters (bytes) from the packet until it finds a '\0', or reaches a
+ * maximum of \c length characters.
+ * When the '\0' has not been reached in the first \c length read characters,
+ * more characters are read from the packet until '\0' has been reached. However,
+ * these characters will not end up in the returned string.
+ * The length of the returned string will be at most \c length - 1 characters.
+ * @param length   The maximum length of the string.
+ * @param settings The string validation settings.
+ * @return The validated string.
+ */
+std::string Packet::Recv_string(size_t length, StringValidationSettings settings)
+{
+	assert(length > 1);
+
+	std::string str;
+	char character;
+	while (--length > 0 && (character = this->Recv_uint8()) != '\0') str.push_back(character);
+
+	if (length == 0) {
+		/* The string in the packet was longer, read until the termination.
+		 * Note that Recv_uint8 returns 0 when reading beyond the packet. */
```

Comment:
how is this "note" helping me here?

Reply:
At a first glance it might look like an infinite loop that could start reading out of bounds, so this hints that it's terminating. Maybe I should have thought of a better way of conveying that information though.